### PR TITLE
Add one-to-one relation

### PR DESCRIPTION
**Pull request by multimikael**
Support one-to-one relations using @ForeignKey on models.

### CODE REVIEW

**Comment by HenrikHL** [comment]
In general I am ok with this PR. I would though have turned the @ForeignKey annotation around and placed it on the object that needs to be "expanded/filled" instead of on the id... This would allow for combined keys as well - I cannot see how that can be done in the current suggestion...

File: src/main/java/org/dcsa/core/repository/RowMapper.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RowMapper {
+    private static final Integer DATABASE_INTERVAL_NATIVE_TYPE = 1186;
+
+    public static <T> T mapRow(Row row, RowMetadata metadata, DBEntityAnalysis<T> dbEntityAnalysis, Class<T> modelClass, boolean ignoreUnknownProperties) {
```
Comment (nt-as) [inline src/main/java/org/dcsa/core/repository/RowMapper.java:26]: I think we should make this "non-static" and initialize the object mapper as field.  Something like:

```java
    private final ObjectMapper objectMapper = new ObjectMapper()
                   .registerModule(new JavaTimeModule()) // Needed to process time objects.
                   .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
                   .disable(DeserializationFeature.ADJUST_DATES_TO_CONTEXT_TIME_ZONE)
                   .enable(SerializationFeature.WRITE_DATES_WITH_ZONE_ID)
                   .setAnnotationIntrospector(new JacksonAnnotationIntrospector() {
                      // Insert body here - too lazy to copy-waste it as multiline copy in GH embeds HTML fragments ...
                   });
```

**Review by nt-as — CHANGES_REQUESTED** [review]

**Comment by nt-as** [comment]
> 
> 
> In general I am ok with this PR. I would though have turned the @foreignkey annotation around and placed it on the object that needs to be "expanded/filled" instead of on the id... This would allow for combined keys as well - I cannot see how that can be done in the current suggestion...

To be honest, I prefer having it here it is and I am discussing the pros/cons with Henrik.

**Comment by nt-as** [comment]
We agreed that the FK annotation is fine where it is.  :)

**Comment by HenrikHL** [comment]
> We agreed that the FK annotation is fine where it is. :)

Yes - we will keep it as it is :-) And expand if needed in the future

**Comment by HenrikHL** [comment]
Before we merge this - Core version needs to be bumped ;-)

**Review by nt-as — APPROVED** [review]

**Review by nt-as — APPROVED** [review]

**Comment by nt-as** [comment]
@HenrikHL: Can you force this one through? Our CI fails because the PR is from an external repo and therefore does not have access to the secret to publish artifacts at the end of the tests.

**Comment by HenrikHL** [comment]
merged...

**Pull request merged**